src/ConnectionAutomaton.cpp

```diff
@@ -119,10 +119,14 @@ void ConnectionAutomaton::disconnect() {
     return;
   }
 
-  closeFrameTransport(folly::exception_wrapper());
-  pauseStreams();
-  stats_.socketDisconnected();
-  onDisconnected_();
+  if (isResumable_) {
```

Comment:
The ConnectionAutomaton::disconnect should not change behavior and close the socket. This method can be called from application code thru ReactiveSocket::disconnect() and the expectation is that it will just disconnect.

I would suggest creating one of the following new methods:
a) disconnectOrCloseWithError(Frame_ERROR)  (more generic approach)
b) keepaliveIntervalFailed(...)    (concrete purpose)